xcdat/climatology.py

```diff
@@ -0,0 +1,355 @@
+"""Functions related to calculating climatology cycles and departures."""
+
+from typing import Union, get_args
+
+import numpy as np
+import xarray as xr
+from typing_extensions import Literal
+
+from xcdat.log import logging
+
+# GROUPS
+# ======
+# Type alias representing available groups for the ``period`` param.
+PeriodGroup = Literal["ANNUALCYCLE", "SEASONALCYCLE", "YEAR"]
```

Comment:
This aligns with CDAT cdutil now. 

I noticed CDAT has seasonal cycle as `SEASONNALCYCLE` instead of `SEASONALCYCLE`.